run.go

```diff
@@ -139,38 +135,102 @@ var commandInspect = cli.Command{
 	Action: actionInspect,
 }
 
-func guessType(filename string) string {
-	switch {
-	case strings.Contains(filename, "://"):
-		return TypeURL
-	case strings.HasSuffix(filename, ".js"):
-		return TypeJS
-	default:
-		return ""
+func looksLikeURL(str []byte) bool {
+	// Try to parse string as URL
+	u, err := url.Parse(strings.TrimSpace(string(str)))
```

Comment:
Sounds reasonable. Especially the part about letting the URL runner parse and generate better error messages. I'll swap it for a regex. But shouldn't we check for "http://" and "https://" rather than any scheme?